### PR TITLE
Handle max question problems in quiz creation

### LINKED ISSUE #5289: Not enough questions freezes the browser during the quiz creation
### Observed behavior

When exercises with not enough questions are selected for the quiz, the browser freezes and needs to be closed.

### Expected behavior

We used to have [an alert message for these cases](https://crowdin.com/translate/kolibri/1002/en-es#70086), but it does not appear to be triggered correctly.

### User-facing consequences
Inability to proceed, forced to restart the browser.

### Errors and logs

```
Not enough questions to reach numQuestions
```

### Steps to reproduce

#. Go to Coach > Plan > Quizzes > New quiz
#. Up the number of questions to 50
#. Select only one exercise
#. Press `Continue`

### Context

0.12.1 (kolibridemo server)

cc @indirectlylit @rtibbles @jonboiser

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Gets rid of unnecessary scrollbars on question items.
image
* Prevents infinite loop when fewer questions have been selected from exercises than required by the quiz.
* Fixes bug where available questions from selected exercises were not being properly calculated.
* Uses available questions to indicate errors when insufficient questions are available to create the quiz.
* Sets the question count for the exercise based on the number of selected questions to prevent a mismatch.

### Reviewer guidance
1. Create a quiz.
2. Select an exercise.
3. Select many more questions to choose than exist in the exercise (e.g. 50).
4. Move onto the next step. Note that an infinite loop does not occur.
5. Notice that regardless of browser, the question list items do not have an unsightly vertical scrollbar.
6. Press 'Finish', note that an error message appears warning you of your folly.
7. Rejoice.

### References
Fixes #5289

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue [hunk 1]
```diff
@@ -37,36 +37,42 @@
...
+                type="flat"
+                aria-hidden="true"
+                class="number-btn"
+                :disabled="numQuestions === maxQs"
+                @click="numQuestions += 1"
+              >
+                <mat-svg name="add" category="content" />
+              </UiIconButton>
+            </KGridItem>
+          </KGrid>
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue]: what was the intent of this change? This is what I'm seeing:

| before | after |
|--|--|
| image |  image |

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue]: The aim was to give the text field a consistent width, as now that the validation messages are showing properly, the `after` view was being shown consistently without this change, but only when the validation text appeared.

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue]: I also only tested this on a wider form factor, so happy for feedback on how to handle this.

**Review by indirectlylit — APPROVED** [review]
new logic looks good.

Would recommend adjusting layout to not allow the number input to become quite so large. Possible solution here: https://github.com/rtibbles/kolibri/pull/17

**Comment by radinamatic** [comment]
Totally after the fact... 😊 

Looking good and triggering well! 👍

**Comment by indirectlylit** [comment]
thanks for testing!

**Pull request merged**